AccountTests/FxAClient10Tests.swift

```diff
@@ -151,4 +151,36 @@ class FxAClient10Tests: LiveAccountTest {
         }
         self.waitForExpectations(timeout: 10, handler: nil)
     }
+    
+    func testProfileSuccess() {
+        withVerifiedAccount { emailUTF8, quickStretchedPW in
+            let e = self.expectation(description: "")
+            
+            let client = FxAClient10()
+            let login: Deferred<Maybe<FxALoginResponse>> = client.login(emailUTF8, quickStretchedPW: quickStretchedPW, getKeys: true)
+            let profile: Deferred<Maybe<FxAProfileResponse>> = login.bind { (result: Maybe<FxALoginResponse>) in
+                switch result {
+                case let .failure(error):
+                    return Deferred(value: .failure(error))
+                case let .success(loginResponse):
+                    return client.getProfile(withSessionToken: loginResponse.value.sessionToken as NSData)
+                }
+            }
+            profile.upon { result in
+                if let response = result.successValue {
+                    XCTAssertNotNil(response.uid)
+                    XCTAssertNotNil(response.email)
+                    
+                    // Avatar and display name are optional user attributes.
+                    // New accounts will not have these set.
+                    XCTAssertNil(response.avatar)
+                    XCTAssertNil(response.displayName)
+                } else {
+                    XCTAssertEqual(result.failureValue!.description, "")
+                }
+                e.fulfill()
+            }
+        }
+        self.waitForExpectations(timeout: 10, handler: nil)
```

Comment:
Don't bother with expectations; use `.succeeded` or `.value` and check `isSuccess`.